controller/src/main/java/io/pravega/controller/server/SegmentHelper.java

```diff
@@ -70,9 +79,15 @@
 
     private final SegmentStoreConnectionManager connectionManager;
     private final HostControllerStore hostStore;
-    
+    private final ConnectionFactory connectionFactory;
+
+    private final Object lock = new Object();
+    @GuardedBy("lock")
+    private RawClient client = null;
```

Comment:
This does not need to be a member variable. All the callers of getConnection use the returned value, so there is no need to store it here.

Reply:
removed